src/components/TodoContent.tsx

```diff
@@ -0,0 +1,244 @@
+            ),
+          ),
+        )
+        .catch(() => setErrorMessage(Errors.notDelete))
+        .finally(() => setComplTodoDeleteLoading([]));
+    });
+  }
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTodoValue(event.target.value);
+
+    if (todoValue !== '') {
+      resetErrorMessage();
+    }
+  }
+
+  function handleDeleteTodo(todoToDelete: Todo) {
+    const filteredTodoses = todosFromServer.filter(
+      filtTodo => filtTodo.id !== todoToDelete.id,
+    );
+
+    setTodosFromServer(filteredTodoses);
+  }
+
+  function handleUpdateTodos(todoses: Todo[]) {
+    setTodosFromServer(todoses);
+  }
+
+  function handleErrorMessage(error: Errors | '') {
+    setErrorMessage(error);
+  }
+
+  function handleAllButtonClick() {
+    todosFromServer.map(todo => {
+      setAnyLoading(true);
+      const updatedTodo: Todo = {
+        ...todo,
+        completed: isAllcompleted ? false : true,
+      };
+
+      updateTodo(updatedTodo)
+        .then(() => getTodos())
+        .then(setTodosFromServer)
+        .catch(() => setErrorMessage(Errors.notUpdate))
+        .finally(() => setAnyLoading(false));
+    });
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    if (todoValue.trim() === '') {
+      setErrorMessage(Errors.notEmpty);
+      setTimeout(resetErrorMessage, 3000);
+
+      return;
+    }
+
+    const optimisticTodo: Todo = {
+      userId: USER_ID,
+      title: todoValue.trim(),
+      completed: false,
+      isOptimistic: true,
+    };
+
+    const { isOptimistic, ...preparedTodo } = optimisticTodo;
+
+    setTodosFromServer(cur => [...cur, optimisticTodo]);
+    setInputDisabled(true);
+
+    postTodo(preparedTodo)
+      .then(responceTodo =>
+        setTodosFromServer(currentTodos =>
+          currentTodos.map(todo => (todo.isOptimistic ? responceTodo : todo)),
+        ),
+      )
+      .then(() => {
+        setTodoValue('');
+        setInputDisabled(false);
+      })
+      .catch(() => {
+        setTodosFromServer(currentTodos =>
+          currentTodos.filter(todo => !todo.isOptimistic),
+        );
+        setErrorMessage(Errors.notAdd);
+        setInputDisabled(false);
+        setTimeout(resetErrorMessage, 3000);
+      })
+      .finally(() => {
+        setLoadingTodo(null);
+      });
+  }
+
+  function setLoadingValue(todo: Todo | null) {
+    setLoadingTodo(todo);
+  }
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setErrorMessage('');
+        setTodosFromServer(data);
+      })
+      .catch(() => {
+        setErrorMessage(Errors.notLoad);
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (todosFromServer.every(todo => todo.completed)) {
+      setIsAllCompleted(true);
+    } else {
+      setIsAllCompleted(false);
+    }
+
+    setCompletedTodos([...todosFromServer].filter(todo => todo.completed));
```

Comment:
it can be var

Reply:
I didn't get it. Whats can be var?

Reply:
```suggestion
   const completedTodos = todosFromServer.filter(todo => todo.completed);
```
<img width="631" alt="image" src="https://github.com/user-attachments/assets/cb00bf02-b604-4a64-a456-314948d2017c">